### PR TITLE
update docs to refer to npm On-Site rather than npm Enterprise

### PR DESCRIPTION
**Pull request by bcoe**
1. updated docs to refer to npm On-Site rather than npm Enterprise.
2. first pass at cleaning up documentation, and making sure that docs better describe the current installation flow.

I still think we'll want to do a few more passes at cleaning up documentation, this simply makes it less blatantly wrong.

### CODE REVIEW

**Comment by bcoe** [comment]
reviewers: @nexdrew, @rod11

File: content/enterprise/client-configuration.md [hunk 1]
```diff
@@ -19,34 +18,38 @@ by running
...
 
 ## Logging in
 
 The `npm login` command will prompt you for your credentials. By default, these will be your GitHub or GitHub Enterprise credentials.
 
-You can configure the server to use different authentication. You can read more about the default authentication configuration and how to configure different authentication.
+You must configure npm On-Site in the admin console to point to your
+GitHub Enterprise server. Read more about our [GitHub Enterprise Integration](/onsite/github).
+
+Read about how to configure custom authentication [here](/onsite/custom-authentication).
```
Comment (nexdrew) [inline content/enterprise/client-configuration.md]: URIs here should be `content/enterprise/github` and `content/enterprise/custom-authentication`, respectively.

File: content/company/security.md [hunk 2]
```diff
@@ -30,7 +30,7 @@ For more information about how we handle your personal data, you may wish to rev
 
 ## Higher Levels of security
 
-For firms interested in greater levels of physical and operational security, [npm Enterprise](/enterprise) is a self-hosted version of the npm Registry that allows total control of the operation and policies of the registry.
+For firms interested in greater levels of physical and operational security, [npm On-Site](/enterprise) is a self-hosted version of the npm Registry that allows total control of the operation and policies of the registry.
```
Comment (nexdrew) [inline content/company/security.md]: URI should be `content/enterprise`

File: content/enterprise/client-configuration.md [hunk 3]
```diff
@@ -19,34 +18,38 @@ by running
...
-npm login --registry=http://myreg.mycompany.com --scope=@myco
+npm login --registry=http://myreg.mycompany.com:8080 --scope=@myco
 ```
 
 ## Logging in
 
 The `npm login` command will prompt you for your credentials. By default, these will be your GitHub or GitHub Enterprise credentials.
 
-You can configure the server to use different authentication. You can read more about the default authentication configuration and how to configure different authentication.
+You must configure npm On-Site in the admin console to point to your
```
Comment (nexdrew) [inline content/enterprise/client-configuration.md]: s/must/should/

File: content/company/security.md (hunk 2 above)
Comment (bcoe) [inline content/company/security.md]: The tool we use to build the docs site actually roots things at `/content`, so `/enteprise` is correct, you were correct that I had broken the prior two links by adding the `onsite` prefix however.

File: content/enterprise/github.md [hunk 4]
```diff
@@ -20,13 +20,18 @@ repository field to your package.json that points to your repo:
 }
 ```
 
-Depending on your configuration, npmE can restrict installation of your
+npm On-Site can restricts installation of your
```
Comment (nexdrew) [inline content/enterprise/github.md]: Either "restricts" or "can restrict", but not "can restricts"

File: content/enterprise/installation.md [hunk 5]
```diff
@@ -4,44 +4,17 @@ title: Installing the server
...
 
 ## Server setup
 
 Follow the instructions in the [quickstart video](/enterprise/intro), or do the following:
 
-1. [Sign up for a trial license](http://www.npmjs.org/enterprise#contact) for npm Enterprise.
-1. Make sure you have a machine that meets the [installation requirements](/enterprise/requirements). It can also be run in a [Docker container](https://github.com/npm/npme-docker).
-1. On the server, run ```npm install npme```. Do NOT run using `sudo`.
+1. [Sign up for a trial license](http://www.npmjs.org/enterprise#contact) for npm On-Site.
+2. Make sure you have a machine that meets the [installation requirements](/enterprise/requirements).
```
Comment (nexdrew) [inline content/enterprise/installation.md]: URI should be `content/enterprise/requirements`

File: content/enterprise/installation.md [hunk 6]
```diff
@@ -4,44 +4,17 @@ title: Installing the server
...
+npm On-Site runs locally, on a server you control, with no external dependencies. Many organizations want this for security, regulatory, or operational reasons.
 
 ## Server setup
 
 Follow the instructions in the [quickstart video](/enterprise/intro), or do the following:
 
-1. [Sign up for a trial license](http://www.npmjs.org/enterprise#contact) for npm Enterprise.
-1. Make sure you have a machine that meets the [installation requirements](/enterprise/requirements). It can also be run in a [Docker container](https://github.com/npm/npme-docker).
-1. On the server, run ```npm install npme```. Do NOT run using `sudo`.
+1. [Sign up for a trial license](http://www.npmjs.org/enterprise#contact) for npm On-Site.
```
Comment (nexdrew) [inline content/enterprise/installation.md]: URL should be `http://www.npmjs.org/onsite#contact`

File: content/enterprise/installation.md [hunk 7]
```diff
@@ -4,44 +4,17 @@ title: Installing the server
...
 
-# Installing the npm Enterprise server
+# Installing the npm On-Site server
 
-npmE runs locally, on a server you control, with no external dependencies. Many organizations want this for security, regulatory, or operational reasons.
+npm On-Site runs locally, on a server you control, with no external dependencies. Many organizations want this for security, regulatory, or operational reasons.
 
 ## Server setup
 
 Follow the instructions in the [quickstart video](/enterprise/intro), or do the following:
```
Comment (nexdrew) [inline content/enterprise/installation.md]: URI `content/enterprise/intro`

File: content/enterprise/installation.md [hunk 8]
```diff
@@ -4,44 +4,17 @@ title: Installing the server
...
 
-1. [Sign up for a trial license](http://www.npmjs.org/enterprise#contact) for npm Enterprise.
-1. Make sure you have a machine that meets the [installation requirements](/enterprise/requirements). It can also be run in a [Docker container](https://github.com/npm/npme-docker).
-1. On the server, run ```npm install npme```. Do NOT run using `sudo`.
+1. [Sign up for a trial license](http://www.npmjs.org/enterprise#contact) for npm On-Site.
+2. Make sure you have a machine that meets the [installation requirements](/enterprise/requirements).
+3. On the server, run ```sudo npm install npmo -g --unsafe```.
+4. visit _http://myreg.mycompany.com:8800_, to configure your instance.
```
Comment (nexdrew) [inline content/enterprise/installation.md]: s/visit/Visit/

File: content/enterprise/intro.md [hunk 9]
```diff
@@ -4,45 +4,60 @@ title: Quickstart
...
+<p>npm On-Site should be installed on a dedicated server.</p>
+<ol> <li>Install the latest version of Node. For Ubuntu, we recommend the distribution put together by our friends at <a href="https://nodesource.com/">NodeSource</a>.<pre><code>curl -sL https://deb.nodesource.com/setup | sudo bash -
 sudo apt-get install nodejs
-</code></pre> </li> <li><code>npm install npme</code> (do NOT use sudo on this command)</li> <li>Answer the questions as prompted. The two you are likely to want to customize are: <ul> <li>The full URL of your front-facing host, followed by the port number npm Enterprise is running on</li> <li>The URL of your GitHub Enterprise appliance. The default for this is public GitHub, which you can use if your company doesn't use GitHub Enterprise.</li> </ul> </li> <li><code>npme start</code></li> </ol>
+</code></pre> </li> <li>next run, <code>sudo npm install npmo -g --unsafe</code></li>
```
Comment (nexdrew) [inline content/enterprise/intro.md]: Capitalize "next"

File: content/enterprise/intro.md [hunk 10]
```diff
@@ -4,45 +4,60 @@ title: Quickstart
...
+<li>now visit <i>http://myreg.mycompany.com:8800</i>, and configure the serving to your liking (configuring authentication, replication, etc).</li>
```
Comment (nexdrew) [inline content/enterprise/intro.md]: Capitalize "now"

File: content/enterprise/intro.md [hunk 11]
```diff
@@ -4,45 +4,60 @@ title: Quickstart
...
+ </ol>
+
+<p><strong>Test:</strong> To make sure that the server is up and running, use <code>curl http://myreg.mycompany.com:8080</code>. You should get a JSON response. This is the same URL that you will login to with npm</p>
```
Comment (nexdrew) [inline content/enterprise/intro.md]: Add period at end of last sentence

File: content/enterprise/intro.md [hunk 12]
```diff
@@ -4,45 +4,60 @@ title: Quickstart
...
 
-<h2>Configuring the client</h2> <a href="https://www.youtube.com/watch?v=EK4fv4iC-4Y#t=144">video</a> <p>Before interacting with your npm Enterprise server from the client, you'll need to update npm and log in.</p>
+ <p>Before interacting with your npm On-Site server from the client, you'll need to update npm and log in.</p>
 
-<ol> <li>Get the latest version of npm. <code>sudo npm install npm -g</code></li> <li><code>npm login --registry="http://example.com/path/to/registry:8080" --scope="@myco"</code></li> <li>Fill in GitHub credentials. If you are using GitHub Enterprise, fill in the credentials that you use on that. Otherwise, use your credentials from public GitHub. Support for other auth services and standalone auth are coming soon.</li> </ol>
+<ol> <li>Get the latest version of npm. <code>sudo npm install npm -g</code></li> <li><code>npm login --registry="http://example.com/path/to/registry:8080" --scope="@myco"</code></li> <li>Fill in GitHub credentials. If you are using GitHub Enterprise, fill in the credentials that you use on that. Otherwise, use your credentials from public GitHub (you may wish to enable an alternative auth-mechanism in The
```
Comment (nexdrew) [inline content/enterprise/intro.md]: Maybe make the fake registry URL consistent with other ones, e.g. `http://myreg.mycompany.com:8080`

File: content/enterprise/intro.md [hunk 13]
```diff
@@ -4,45 +4,60 @@ title: Quickstart
...
+
+<p>If you want to, you can configure npm to only use the private registry. This can be helpful if you want to limit your organization to a certain subset of modules. For example, you might want to only use the subset of modules which have a license which is compatible with your application's license.</p> <ol> <li>In your <code>.npmrc</code> file, add a line <code>registry=http://example.com/path/to/registry:8080</code></li> </ol> <p><strong>Test: </strong>Using the package.json from above, try running <code>npm install</code>. You should get an error that says nodeunit could not be found.</p> <h2>Mirroring the public registry (optional)</h2>
```
Comment (nexdrew) [inline content/enterprise/intro.md]: Possibly needs consistent fake registry URL

File: content/enterprise/intro.md [hunk 14]
```diff
@@ -4,45 +4,60 @@ title: Quickstart
...
+<p>You can optionally mirror modules from public npm. This can be done manually by adding packages to the whitelist, or automatically
+by enabling npm On-Site's read-through-cache in the admin console. To add packages to the whitelist, simply:
 
-<p>You can optionally mirror modules from public npm. This can be done manually by adding packages to the whitelist, or automatically (which is not covered in this tutorial).</p> <ol> <li>On your server, run <code>npme add-package packagename</code>. This will trigger mirroring of the package and its dependencies.</li> </ol> <p><strong>Test: </strong>Look in <code>/etc/npme/packages</code> to see what packages have been mirrored.</p>
+1. login to your npm On-Site appliance.
+2. run `npmo add-package package-name`.
```
Comment (nexdrew) [inline content/enterprise/intro.md]: Maybe capitalize "login" and "run" here.

Should we clarify that "login" means "ssh into"?

File: content/enterprise/mirroring.md [hunk 15]
```diff
@@ -11,32 +11,32 @@ Sometimes you want your private registry to maintain copies of packages from the
 - To ensure developers and build machines can keep working even when there is a connectivity problem
 - To maintain an independent copy of all package versions, in case an author unpublishes a version
 
-You can set up different policies to dictate how your npm Enterpise server manages mirroring.
+You can set up different policies to dictate how your npm On-Site server manages mirroring.
 
-NOTE: If you want clients to use the mirrored packages from your npm Enterprise server instead of accessing them from the public registry, the clients must be [configured](/enterprise/client-configuration#using-your-private-registry-for-all-packages) to do so.
+NOTE: If you want clients to use the mirrored packages from your npm On-Site server instead of accessing them from the public registry, the clients must be [configured](/enterprise/client-configuration#using-your-private-registry-for-all-packages) to do so.
```
Comment (nexdrew) [inline content/enterprise/mirroring.md]: Hopefully I'm not messing up all the links once this is deployed, but this URI should be `content/enterprise/client-configuration.md#using-your-private-registry-for-all-packages` to work in GitHub.

File: content/enterprise/no-authentication.md [hunk 16]
```diff
@@ -5,21 +5,13 @@ title: No authentication
...
-to `"fake"`
-  * `args.--session-handler` to `"redis"`
-
-Make sure to regenerate scripts as described in
-[Customizing server configuration](/enterprise/server-configuration).
+In order to do so, visit npm On-Site's admin console, and choose the `Open` authentication strategy.
 
 The npm CLI will still require that you log in in order to publish
-packages, but you can just log in with bogus credentials (although the
+packages, but you can just log in with any credentials (although the
```
Comment (nexdrew) [inline content/enterprise/no-authentication.md]: I assume npm On-Site will restrict the number of users that are allowed to authenticate to the number they are licensed for? If so, perhaps we should mention that here?

File: content/enterprise/requirements.md [hunk 17]
```diff
@@ -3,16 +3,15 @@ order: 2
 title: Requirements
 -->
 
-# npm Enterprise installation requirements
+# npm On-Site installation requirements
 
-npm Enterprise has been extensively tested Ubuntu 14 (Trusty), and is
-supported on most recent public Linux distributions. It can run inside a
-virtual machine or on standard hardware. It can also be run in a [Docker container](https://github.com/npm/npme-docker).
+npm On-Site has been extensively tested Ubuntu 14 (Trusty), and is
```
Comment (nexdrew) [inline content/enterprise/requirements.md]: s/tested/tested on/

File: content/enterprise/updating.md [hunk 18]
```diff
@@ -1,13 +1,8 @@
...
-## Updating npm Enterprise
+## Updating npm On-Site
 
-npm Enterprise can self-update. To ensure that all dependencies are updated, remove the @npm directory before running the update.
-
-```bash
-rm -rf /etc/npme/node_modules/@npm
-npme update
-```
+to update npm On-Site, simply click `Check Now` in the admin console.
```
Comment (nexdrew) [inline content/enterprise/updating.md]: Maybe capitalize "to"

**Comment by nexdrew** [comment]
Sorry about breaking the links. Stinks that they can't work in GitHub *and* on the web site.

I'm done with my stupid nits. This is looking much better already! :+1:

**Comment by bcoe** [comment]
\o/ @nexdrew thanks for the review!

**Pull request merged**